### ISSUE TITLE
Scalaz 8 Maybe does not have exaustiveness warnings for pattern matches.

### ISSUE THREAD

**Issue by Jacoby6000** [scalaz/scalaz#1832]
I suspect this is true for other newtype sums that we have.

**Comment by jbgi** [comment]
Yes, this a known issue (cf. comments in thread https://github.com/scalaz/scalaz/pull/1450#issuecomment-381231202), but it is good to have a dedicated github issue to track it.

Personally I'd be in favor of removing the extractors (and use pattern match on `toOption` when needed), until we have a solution that preserve exhaustive checks.
